### PR TITLE
Java2063 Normalize authentication error logging.

### PR DESCRIPTION
**Pull request by GregBestland**
This brings control connection Authentication logging inline with the channel pool authentication logging. They are both logged at warn.

### CODE REVIEW

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java [hunk 1]
```diff
@@ -501,6 +515,20 @@ private void forceClose() {
     }
   }
 
+  private boolean isAuthFailure(Throwable error) {
+    boolean authFailure = true;
+    if (error instanceof AllNodesFailedException) {
+      Collection<Throwable> errors = ((AllNodesFailedException) error).getErrors().values();
```
Comment (tomekl007) [inline core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java:561]: if `getErrors()` will return empty connection we will treat it as the `AuthFailure`. Is it ok for us? Is it possible?

Comment (GregBestland) [reply core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java:561]: That SHOULD never happen, I'll add a check regardless.

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java [hunk 2]
```diff
@@ -273,6 +275,13 @@ private void init(boolean listenToClusterEvents, boolean reconnectOnFailure) {
               firstConnectionAttemptFuture.complete(null);
             },
             error -> {
+              if (isAuthFailure(error)) {
+                Loggers.warnWithException(
+                    LOG,
+                    "[{}] Authentication errors encountered on all contact points. Please check your authentication configuration.",
+                    logPrefix,
+                    error);
+              }
```
Comment (olim7t) [inline core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java:311]: The errors have already been logged individually in `connect`, do we need another log here?
For example run `PlainTextAuthProviderIT.should_not_connect_without_credentials()`, there are two warning logs.

If we decide to keep this one, we should at least not include the exception, the stack trace doesn't bring any useful information.

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java [hunk 3]
```diff
@@ -332,11 +341,16 @@ private void connect(
...
-                            error);
+                        if (error instanceof AuthenticationException) {
+                          Loggers.warnWithException(
+                              LOG, "[{}] Authentication error", logPrefix, error);
+                        } else {
+                          LOG.debug(
+                              "[{}] Error connecting to {}, trying next node",
+                              logPrefix,
+                              node,
+                              error);
```
Comment (olim7t) [inline core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java]: Could you also honor `DefaultDriverOption.CONNECTION_WARN_INIT_ERROR` here?
See `ChannelPool.onAllConnected()`.

**Review by olim7t — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java (hunk 2 above)
Comment (GregBestland) [reply core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java:311]: I'd like to keep this to be explicit to the user that something about their configuration is probably wrong. I have no problem removing the exception and stack trace.

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java (hunk 3 above)
Comment (GregBestland) [reply core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java]: I think that should address your issues. If Feel very strongly about the cumulative logging warning I can remove it, I just want to be explicit about their configuration being incorrect.

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java (hunk 2 above)
Comment (olim7t) [reply core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java:311]: Fair enough, there could be auth errors on _some_ nodes but not all, the cumulative warning helps distinguish that case.

**Comment by olim7t** [comment]
LGTM. I had a couple more trivial comments, I addressed them directly.

Last thing: tests are failing, you just need to mock the config option in `ControlConnectionTestBase`. After that this is good to go.

**Review by olim7t — APPROVED** [review]

**Comment by olim7t** [comment]
Nit: could you use the usual conventions in the PR title next time? When you merge with "squash and merge", that's what GitHub uses for the commit message.
```
d6cb615e8 - (datastax/4.x) JAVA-2119: Add PagingIterable abstraction
9a81e4470 - Java2063 Normalize authentication error logging. (#1176)
a2af2edcc - JAVA-2084: Fix session.executeAsync examples in the docs
```
That's a small detail, but it's convenient in case you need to grep a ticket number in the log.

**Pull request merged**